service/src/worker.rs

```diff
@@ -77,13 +77,10 @@ where
 			// FIXME: Websocket connectionto a worker  should stay once etablished.
 			let client = WsClientBuilder::default().build(&url).await?;
 			let blocks = blocks_json.clone();
-			thread::spawn(move || {
-				if let Err(e) = futures::executor::block_on(
-					client.request::<Vec<u8>>("sidechain_importBlock", blocks.into()),
-				) {
-					error!("sidechain_importBlock failed: {:?}", e);
-				}
-			});
+			if let Err(e) = client.request::<Vec<u8>>("sidechain_importBlock", blocks.into()).await
+			{
+				error!("sidechain_importBlock failed: {:?}", e);
```

Comment:
is there a reason for snake_andCamel_case?

Reply:
It's an json rpc call, with a function call named like that.

Reply:
And the json rpc calls were named according to the example of substrate: https://docs.substrate.io/v3/runtime/custom-rpcs/#public-rpcs

Reply:
To answer your question: No I don't think there is a good reason. The idea behind it was probably that the first part is like a prefix/namespace, followed by a function name that is in camel case (as is more common in RPC function names, because languages like Java and Javascript more commly use camel and pascal case for function names).

Reply:
Following the example of substrate is a good reason I think, because people working with our worker most likely know substrate, so they are already used to this naming convention.

Reply:
interesting. thanks.

Reply:
Yes, substrate has this specific naming, as they needed a means for auto-generating rpcs for modules and their calls.

So for every module, the `#[rpc]` proc-macro generates rpcs that are like `module_fnName`

Reply:
oho. So that's why. Thanks for the even better explanation @clangenb 👍